test_run.go

```diff
@@ -0,0 +1,328 @@
+type TestRunCreateOptions struct {
+	// Type is a public field utitilized by JSON:API to set the resource type
+	// via the field tag. It is not a user-defined value and does not need to
+	// be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,test-runs"`
+
+	// If non-empty, requests that only a subset of testing files within the
+	// ConfigurationVersion should be executed.
+	Filter []string `jsonapi:"attr,filter,omitempty"`
+
+	// Specifies the directory within the ConfigurationVersion that test files
+	// should be loaded from. Defaults to "tests" if empty.
+	TestDirectory *string `jsonapi:"attr,test-directory,omitempty"`
+
+	// Verbose prints out the plan and state files for each run block that is
+	// executed by this TestRun.
+	Verbose *bool `jsonapi:"attr,verbose,omitempty"`
+
+	// Variables allows you to specify terraform input variables for
+	// a particular run, prioritized over variables defined on the workspace.
+	Variables []*RunVariable `jsonapi:"attr,variables,omitempty"`
+
+	// ConfigurationVersion specifies the configuration version to use for this
+	// test run.
+	ConfigurationVersion *ConfigurationVersion `jsonapi:"relation,configuration-version"`
+
+	// RegistryModule specifies the registry module this test run should be
+	// assigned to.
+	RegistryModule *RegistryModule `jsonapi:"relation,registry-module"`
+}
+
+// TestRunList represents a list of test runs.
+type TestRunList struct {
+	*Pagination
+	Items []*TestRun
+}
+
+// TestRunListOptions represents the options for listing runs.
+type TestRunListOptions struct {
+	ListOptions
+}
+
+// List all the test runs for a given private registry module.
+func (s *testRuns) List(ctx context.Context, moduleID RegistryModuleID, options *TestRunListOptions) (*TestRunList, error) {
+	if err := moduleID.valid(); err != nil {
+		return nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", testRunsPath(moduleID), options)
+	if err != nil {
+		return nil, err
+	}
+
+	trl := &TestRunList{}
+	err = req.Do(ctx, trl)
+	if err != nil {
+		return nil, err
+	}
+
+	return trl, nil
+}
+
+// Read a test run by its ID.
+func (s *testRuns) Read(ctx context.Context, moduleID RegistryModuleID, testRunID string) (*TestRun, error) {
+	if err := moduleID.valid(); err != nil {
+		return nil, err
+	}
+
+	if !validStringID(&testRunID) {
+		return nil, ErrInvalidTestRunID
+	}
+
+	u := fmt.Sprintf("%s/%s", testRunsPath(moduleID), url.QueryEscape(testRunID))
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	tr := &TestRun{}
+	err = req.Do(ctx, tr)
+	if err != nil {
+		return nil, err
+	}
+
+	return tr, nil
+}
+
+// Create a new test run with the given options.
+func (s *testRuns) Create(ctx context.Context, options TestRunCreateOptions) (*TestRun, error) {
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	moduleID := RegistryModuleID{
+		Organization: options.RegistryModule.Organization.Name,
```

Comment:
Organization _could_ be nil here, causing a panic. You could check for this in TestRunCreateOptions valid

Reply:
Done!